### PR TITLE
Replace $BUILD_PREFIX in generated config files

### PR DESCRIPTION
**Pull request by abretaud**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

Here's an attempt at fixing #29 
The recipe builds successfully on my laptop, but i haven't yet tested if the change is enough to fix the problem when building other perl modules
I'm not an expert of this, so I hope this won't break anything else...

### CODE REVIEW

**Comment by abretaud** [comment]
@conda-forge-admin, please rerender

**Comment by abretaud** [comment]
Tested and... it fails...

**Comment by abretaud** [comment]
@conda-forge-admin, please rerender

**Comment by abretaud** [comment]
ping @conda-forge/core
No idea why rerender failed

**Comment by abretaud** [comment]
So with the latest change, instead of replacing the hard-coded paths by other hard-coded paths, I made the perl code look at the env var to guess the correct paths. Not using PREFIX or BUILD_PREFIX as we we don't know if we're building a conda package of just using in normal life.

I tested it by installing locally this perl modified package + perl-extutils-makemaker into a clean conda env, then trying to build [perl-statistics-caseresampling package](https://github.com/bioconda/bioconda-recipes/pull/13713/files#diff-5d53a9e26177da5e01b532fee95756b7) with `perl Makefile.PL INSTALLDIRS=site && make`. And this time it worked.

I don't know if there might be other cases where it won't work, but I think it's already better

**Comment by abretaud** [comment]
I think the appveyor is red only because of network problem

**Comment by abretaud** [comment]
@conda-forge-admin, please rerender

**Comment by bgruening** [comment]
@conda-forge-admin, please rerender

**Comment by abretaud** [comment]
ICYMI ping @jakirkham @mingwandroid @msarahan

File: recipe/build.sh [hunk 1]
```diff
@@ -51,3 +51,16 @@ chmod -R o-w "${SRC_DIR}"
 # https://rt.perl.org/Public/Bug/Display.html?id=128020
 # make test
 make install
+
+# Replace hard-coded BUILD_PREFIX by value from env as CC, CFLAGS etc need to be properly set to be usable by ExtUtils::MakeMaker module
+sed -i.bak "s|\(local \*.*\)|use File::Basename; my \$compilerroot = dirname(\$ENV{'CC'}).\"\/..\/\"; \1|g" $PREFIX/lib/*/*/Config_heavy.pl
```
Comment (bgruening) [inline recipe/build.sh]: @abretaud I know its more work, but can we use a patch here, I think this would be more readable.

Comment (abretaud) [reply recipe/build.sh]: Hum, yep it should be possible for the 2 first sed commands, for the others it's modifying variable content so I'll have to leave them
I'll look at updating to 5.26.3 too
thanks for looking at it!

**Comment by abretaud** [comment]
Changes done!
If updating to 5.26.3, does it mean all perl-* packages will need to be rebuilt?

**Comment by druvus** [comment]
The perl pinning is set to `max_pin: x.x.x` so maybe better to fix 5.26.2 before updating?

**Comment by abretaud** [comment]
Ok, yep I prefer fixing 5.26.2 first too

**Comment by abretaud** [comment]
Any chance to get this merged soon? It's blocking a few things in my todo list

**Comment by abretaud** [comment]
Super cool, thanks @bgruening!

**Comment by abretaud** [comment]
Ah. There's a little problem with my patch (https://circleci.com/gh/bioconda/bioconda-recipes/51272)... I'm working on a fix

**Pull request merged**